### PR TITLE
Trial pypy test suite

### PR DESCRIPTION
**Pull request by alanmcruickshank**
This is a testing PR to attempt testing and building of sqlfluff on pypy, with the intent of more officially supporting pypy as an option in performance work. Once testing is set up and tests pass, this PR should eventually also include sufficient documentation on how to run sqlfluff on pypy.

### CODE REVIEW

**Comment by barrywhart** [comment]
Looks like CircleCI was down, causing the build to fail.

File: tox.ini [hunk 1]
```diff
@@ -9,7 +9,14 @@ allowlist_externals =
...
     -Urrequirements.txt
-    -Urrequirements_dev.txt
+    # All test cases require pytest
+    pytest
+    pytest-cov
+    pytest-sugar
+    hypothesis
+    coverage
+    # If we're in pypy don't install all of these dependencies, otherwise grab the full set.
+    !pypy37: -Urrequirements_dev.txt
```
Comment (barrywhart) [inline tox.ini:19]: Do you know why it's necessary to exclude PyPy here?

Comment (alanmcruickshank) [reply tox.ini:19]: `mypy` doesn't support `pypy`. It's because the dependency `typed_ast` doesn't support `pypy`.

**Comment by alanmcruickshank** [comment]
Based on automated testing taking ages, and pypy not performing that well I'm going to abandon this.

**Pull request closed without merging**